x-pack/plugins/triggers_actions_ui/public/application/components/simple_connector_form.tsx

```diff
@@ -77,39 +94,55 @@ const getFieldConfig = ({
       : []),
   ],
   defaultValue,
+  ...(type && FIELD_TYPES[type]
+    ? { type: FIELD_TYPES[type], defaultValue: Array.isArray(defaultValue) ? defaultValue : [] }
+    : {}),
 });
 
 const FormRow: React.FC<FormRowProps> = ({
   id,
   label,
   readOnly,
   isPasswordField,
+  isRequired = true,
   isUrlField,
   helpText,
   defaultValue,
+  euiFieldProps = {},
+  type,
 }) => {
   const dataTestSub = `${id}-input`;
+  let UseField = UseTextField;
+  if (type && FIELD_TYPES[type] === FIELD_TYPES.COMBO_BOX) {
+    UseField = UseComboBoxField;
+  }
```

Comment:
Can we use a function to get the component? For example:

```
const getComponentForType = (type: FieldType) => {
switch (type) {
    case FIELD_TYPES.COMBO_BOX:
      return getUseField({ component: ComboBoxField });
    default:
       return getUseField({ component: Field });
}
}
```

This way we can extend it easier in the future if needed.